src/lib/api.ts

```diff
@@ -1,12 +1,43 @@
-import type { Post } from '../types';
+import type { DualPost, ApiPost } from '../types';
 import { bucketSlug, readKey } from './config';
 
-export async function getAllPosts(): Promise<Post[]> {
+export async function getAllPosts(): Promise<DualPost[]> {
   const endpoint = `https://api.cosmicjs.com/v2/buckets`;
-  const query = encodeURIComponent(`{"type":"posts"}`);
+  const type = import.meta.env.MODE === 'development' ? 'test-posts' : 'posts';
+  const query = encodeURIComponent(`{"type":"${type}"}`);
   const url = `${endpoint}/${bucketSlug}/objects?&query=${query}&read_key=${readKey}`;
   const response = await fetch(url);
   const objs = await response.json();
-  const allPosts: Array<Post> = objs.objects;
-  return allPosts;
+  const allApiPosts: Array<ApiPost> = objs.objects;
+  return allApiPosts.map(toDual);
+}
+
+function toDual(api: ApiPost): DualPost {
```

Comment:
this is where I turn the API's raw output into our own, more ergonomic `DualPost` type